quilter.py

```diff
@@ -227,15 +227,13 @@ def set_gdal_options(file_list):
     return options
 
 
-def raster_project(raster_folder, extensions, crs):
+def raster_project(raster_folder, out_dir, extensions, crs):
     '''
     Reprojects any rasters (defined by comparing file names to extensions) in raster_folder to a new directory in the same directory as raster_folder (eg, will create ../foo/projected for rasters in ../foo/rasters). crs must be in the form EPSG:xxxx or ESRI:xxxx.
     '''
 
     #: Create output directory
-    parent_dir = os.path.split(raster_folder)[0]
-    projected_dir = os.path.join(parent_dir, 'projected')
-    os.mkdir(projected_dir)
+    os.mkdir(out_dir)
```

Comment:
needs a try catch or a exists call to guard against FileExistsError. You also might consider using makedirs incase its a nested folder structure that doesn't exist yet.

Reply:
Quilter does these file checks towards the beginning of main() (lines 440-461)- a main output directory is created via makedirs() if it doesn't exist already, then the sub directories are checked to see if they exist.

Doing the check early is probably not the most pythonic way or the proper way of doing it. While it's not multi-threaded/process, a race-condition-like situation could occur that the directory structure is modified in between the check and when the program uses the directories. From a user-friendliness standpoint, I want to make sure that any file-related errors are caught and printed before the potentially large time commitment of downloading several large files. 

I'm modifying the program to better use context-managed temp dirs when appropriate and to create the sub directories immediately after checking, rather than waiting until getting to the specific step in the program.

Reply:
great. That is a great improvement. Why aren't you cleaning up the temp dir any longer? Does python do that for you?

Reply:
Yeah- TemporaryFile() is used as context manager (`with tempfile.TemporaryFile() as temp_dir:`), so temp_dir is always cleaned up at the end of the block, even if an exception is thrown. The downside is that I have to be careful to not go out of scope before I'm done using any files stored in temp_dir.